File: LeetCode/src/main/java/com/jack/hashtable/LeetCode219.java
package com.jack.hashtable;

import java.util.HashMap;
import java.util.Map;

public class LeetCode219 {
    public static void main(String[] args) {
        
    }


    static class Solution{
        public boolean containsNearbyDuplicate(int[] nums, int k) {

            int len = nums.length;

            Map<Integer, Integer> map = new HashMap<>();
            for (int i = 0; i < len; i++) {
                int num = nums[i];
                if (map.containsKey(num) && (i-map.get(num)<=k)) return true;

                map.put(num, i);
            }

            return false;
        
        }
    }
}

File: LeetCode/src/main/java/com/jack/hashtable/LeetCode219.py
from typing import List


class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        mp = {}
        
        for i, v in enumerate(nums):
            if v in mp and i-mp[v]<=k:
                return True
            
            mp[v]= i

        return False

if __name__ == "__main__":
    
    nums = [1,2,3,1]
    k = 3
    res = Solution().containsNearbyDuplicate(nums, k)
    
    print (res)